bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/EntityProvider.java

```diff
@@ -29,61 +31,76 @@
 import javax.ws.rs.ext.MessageBodyWriter;
 import javax.ws.rs.ext.Provider;
 
+import org.eclipse.smarthome.io.json.JsonBindingService;
 import org.osgi.service.component.annotations.Component;
-
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
+import org.osgi.service.component.annotations.Reference;
 
 /**
- * JSON reader/writer for Jersey using GSON.
+ * JSON reader/writer for JAX-RS that uses {@link JsonBindingService} to serialize and deserialize between
+ * Java objects and JSON.
  *
  * @author Simon Kaufmann - initial contribution and API
+ * @author Flavio Costa - Renamed from GsonProvider to EntityProvider, removed Gson dependencies
  *
- * @param <T>
+ * @param <T> Type to be serialized or deserialized.
  */
 @Provider
 @Produces(MediaType.APPLICATION_JSON)
 @Consumes(MediaType.APPLICATION_JSON)
-@Component(immediate = true)
-public class GsonProvider<T> implements MessageBodyReader<T>, MessageBodyWriter<T> {
+@Component(immediate = true, service = { MessageBodyWriter.class, MessageBodyReader.class })
```

Comment:
the services are provided implicitly, no need to specify them explicitly.

Reply:
IMHO it is okay for the `service` attribute to state it explicit. But I have no strong meaning.

Reply:
If I remember correctly, I added these not because of OSGi, but because Jersey was not picking up the provider, but I can test again to confirm...